util/EIP-2124/src/lib.rs

```diff
@@ -0,0 +1,314 @@
+// Copyright 2015-2019 Parity Technologies (UK) Ltd.
+// This file is part of Parity Ethereum.
+
+// Parity Ethereum is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Parity Ethereum is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Parity Ethereum.  If not, see <http://www.gnu.org/licenses/>.
+
+//! EIP-2124 implementation based on <https://eips.ethereum.org/EIPS/eip-2124>
+
+use crc::crc32;
+use ethereum_types::H256;
+use maplit::*;
+use rlp::{DecoderError, Rlp, RlpStream};
+use std::collections::{BTreeMap, BTreeSet};
+
+pub type BlockNumber = u64;
+
+#[derive(Clone, Copy, Debug, PartialEq, Eq, Hash)]
+pub struct ForkHash(pub u32);
```

Comment:
all pub types and field should have some docs

Reply:
…and I think we should have `#[deny(missing_docs)]` on all new code.

Reply:
done